api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/IncomingConnectorFactory.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2018-2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.reactive.messaging.spi;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+import javax.enterprise.inject.spi.DeploymentException;
+import java.util.NoSuchElementException;
+
+/**
+ * SPI used to implement a connector managing a source of messages for a specific <em>transport</em>. For example, to
+ * handle the consumption of records from Kafka, the reactive messaging extension would need to implement a {@code bean}
+ * implementing this interface. This bean is called for every {@code channel} that needs to be created for this specific
+ * <em>transport</em> (so Kafka in this case). These channels are connected to methods annotated with
+ * {@link org.eclipse.microprofile.reactive.messaging.Incoming}.
+ * <p>
+ * The factory is called to create a {@code channel} for each configured <em>transport</em>. The configuration is done using
+ * MicroProfile Config. The following snippet gives an example for a hypothetical Kafka connector:
+ *
+ * <pre>
+ * mp.messaging.incoming.my-channel.type=i.e.m.reactive.messaging.impl.kafka.Kafka
+ * mp.messaging.incoming.my-channel.bootstrap-servers=localhost:9092
+ * mp.messaging.incoming.my-channel.topic=my-topic
+ * ...
+ * </pre>
+ * <p>
+ * The configuration keys are structured as follows: {@code mp.messaging.[incoming|outgoing].channel-name.attribute}.
+ * <p>
+ * <p>
+ * The portion of the key that precedes the {@code attribute} is acts as a property prefix that has a common structure
+ * across all {@code channels}.
+ * </p>
+ * <p>
+ * The {@code channel-name} segment in the configuration key corresponds to the name of the channel used in the
+ * {@code Incoming} annotation:
+ *
+ * <pre>
+ * &#64;Incoming("my-channel")
+ * public void consume(String s) {
+ *      // ...
+ * }
+ * </pre>
```

Comment:
<p>
The _connector-qualifier-name_ segment in the configuration key corresponds to the classname of the parameter passed to the {@code Connector} annotation that is used as a CDI qualifier in connector resolution. 
 *
 * <pre>
 * &#64;Connector(com.acme.Kafka.class)
 * &#64;Incoming("my-channel")
 * public void consume(String s) {
 *      // ...
 * }
 * </pre>

The above connector could be overridden with a property of, for example:
```
 mp.messaging.incoming.my-channel.connector=io.moresql.Rfewdbc
```
This would be equivalent of `@Connector( io.moresql.Rfewdbc.class)`

Reply:
My previous comment about using String instead of class would transform this to:

```
@Connector("acme.kafka")
```

and

```
mp.messaging.incoming.my-channel.connector=acme.kafka
```

Note that it would also impact connector _global_ config:

```
mp.messaging.acme.kafka.attr=value
```